grails-app/services/icaruswings/CustomerService.groovy

```diff
@@ -50,6 +51,43 @@ class CustomerService {
         return customer
     }
 
+    public void update(CustomerAdapter customerAdapter) {    
+        if (!CheckEntityExistenceById.CheckCustomerExistenceById(customerAdapter.id)) throw new RuntimeException("Cliente nao encontrado")    
+
+        Customer validatedCustomer = validateDefaultFields(customerAdapter)
+
+        if (validatedCustomer.hasErrors()) {
+            throw new ValidationException("Não foi possível salvar o cliente", validatedCustomer.errors)
+        }
+
+        Long id = Long.parseLong(customerAdapter.id)
+        Customer customer = Customer.get(id)
+
+        customer.name = customerAdapter.name
+
+        if (customer.email != customerAdapter.email) {
+            customer.email = customerAdapter.email
+        }
```

Comment:
?

Reply:
Entendo a ideia de só ataualizar se tiver sido alterado. 
Mas se for seguir essa lógica, isso não deveria ser aplicado pra todos os campos?

Reply:
PS: Não precisa aplicar essa vlaidação pra todos os campos.

Pode fazer a atribuição direta mesmo. Se o valor não tiver sido alterado, já vai ficar tudo igual